### ISSUE TITLE
Understand how to build RL environment.

### ISSUE THREAD

**Issue by x-tu** [x-tu/Fair-RL#2]
# Introduction
Stable-Baselines3 provides open-source implementations of deep reinforcement learning (RL) algorithms in Python. 
1. [Official website](https://jmlr.org/papers/v22/20-1364.html#:~:text=Stable%2DBaselines3%20provides%20open%2Dsource,cover%2095%25%20of%20the%20code.)
2. [Paper](https://jmlr.org/papers/volume22/20-1364/20-1364.pdf)
3. [Github docs](https://stable-baselines3.readthedocs.io/en/master/guide/examples.html)

## History
OpenAI Gym: environments
OpenAI Baselines: high-quality DRL Agents
`Stable-baselines` is based on [OpenAI Baselines](https://github.com/openai/baselines).

## Compatibility
1. stable-baselines is based on `tensorflow 1.x` 
2. stable-baselines3 is based on `pytorch`

# Customized Environments
> Refer to [1]
My github repo: [Fair-RL](https://github.com/x-tu/Fair-RL)
Endangered species - Example: [FDRL](https://proceedings.mlr.press/v119/siddique20a.html)
inherit from `gym.Env`

- [x] [A `callback` function is required to implement a customized training process (CH)](https://zhuanlan.zhihu.com/p/348303193)
- [x] [(Vectorized Environments)](https://stable-baselines.readthedocs.io/en/master/guide/vec_envs.html#wrappers) (Base class for multiple environments, leave for later.)

## [Step](https://github.com/openai/gym/blob/master/gym/core.py#L42) (Core Control)
+ Input: an action from the `self.action_space`
-  Output: `( observation, reward, done, info )`
	- `observation`: from `self.observation_space`
	- `reward`: float
	- `done`: bool, whether the episode ends
	- `info`: dictionary, help debugging

## [Spaces](https://github.com/openai/gym/tree/master/gym/spaces) (Core Data)
* spaces are defined from  [gym.spaces](https://github.com/openai/gym/tree/master/gym/spaces)
* two elements: `observation_space`  and `action_space`
* the upper and lower limits: `low`  and  `high`

## [Reset ](https://github.com/openai/gym/blob/master/gym/core.py#L60)
* initialization for each `episode`
* Must return an initial observation data specified by `self.observation_space`

## [Render](https://github.com/openai/gym/blob/master/gym/core.py#L68)
-  for environment visualization
- override if do not need (may raise `_NotImplementedError_`)
